api/v1alpha3/azurecluster_conversion.go

```diff
@@ -58,6 +58,20 @@ func (src *AzureCluster) ConvertTo(dstRaw conversion.Hub) error { // nolint
 	dst.Spec.NetworkSpec.APIServerLB.FrontendIPsCount = restored.Spec.NetworkSpec.APIServerLB.FrontendIPsCount
 	dst.Spec.NetworkSpec.NodeOutboundLB = restored.Spec.NetworkSpec.NodeOutboundLB
 
+	for i, restoredSubnet := range restored.Spec.NetworkSpec.Subnets {
+		for j, dstSubnet := range dst.Spec.NetworkSpec.Subnets {
+			if dstSubnet.Name == restoredSubnet.Name {
+				for k, restoredSecurityRule := range restoredSubnet.SecurityGroup.SecurityRules {
+					for l, dstSecurityRule := range dstSubnet.SecurityGroup.SecurityRules {
+						if dstSecurityRule.Name == restoredSecurityRule.Name {
+							dst.Spec.NetworkSpec.Subnets[j].SecurityGroup.SecurityRules[l].Direction = restored.Spec.NetworkSpec.Subnets[i].SecurityGroup.SecurityRules[k].Direction
+						}
+					}
+				}
```

Comment:
ditto: break here?

Reply:
Not sure if it'll make a huge difference. I'm fine with skipping it, if it's going to make the code more complex since we'll need to assign the loop to a variable to break.

Reply:
Looking at it again, I think it should be a straightforward break since we are just breaking the inner loop. Sorry about the confusion.

Reply:
added